turbopack/crates/turbopack/src/lib.rs

```diff
@@ -190,10 +190,15 @@ async fn apply_module_type(
                 let options = options.await?;
                 match options.tree_shaking_mode {
                     Some(TreeShakingMode::ModuleFragments) => {
-                        Vc::upcast(EcmascriptModulePartAsset::select_part(
+                        let part_module = EcmascriptModulePartAsset::select_part(
                             module,
                             part.unwrap_or(ModulePart::facade()),
-                        ))
+                        );
+                        if let Some(part_module) = *part_module.await? {
+                            Vc::upcast(part_module)
+                        } else {
+                            return Ok(ProcessResult::Ignore.cell());
```

Comment:
That's great. We can use that for ModuleEvaluation too if that is empty. So that we don't generate a module evaluation fragment in these cases at all.